context.go

```diff
@@ -187,22 +192,7 @@ func (c *Context) reset(view byte) {
 		}
 	}
 
-	data, err := c.Config.Pub.MarshalBinary()
-	if err != nil {
-		panic("can't marshal config public key")
-	}
-
-	id := fetchID(data)
-
-	for i := range c.Validators {
-		data, err := c.Validators[i].MarshalBinary()
-		if err != nil {
-			panic("can't marshal public key")
-		} else if id == fetchID(data) {
-			c.MyIndex = i
-			break
-		}
-	}
+	c.MyIndex, c.Priv, c.Pub = c.Config.GetKeyPair(c.Validators)
```

Comment:
I'm not sure what happens after this invocation when `c.MyIndex == -1`?

Reply:
We will only accept payloads and advance state machine, without creating new messages.
https://github.com/nspcc-dev/dbft/blob/master/context.go#L88